lib/internal/test_runner/runner.js

```diff
@@ -181,6 +185,26 @@ function runTestFile(path, root, inspectPort, filesWatcher) {
   return subtest.start();
 }
 
+function watchFiles(testFiles, root, inspectPort) {
+  const filesWatcher = new FilesWatcher({ throttle: 500, mode: 'filter' });
+  filesWatcher.on('changed', ({ owners }) => {
+    filesWatcher.unfilterFilesOwnedBy(owners);
+    const dependantTests = ArrayPrototypeFilter(testFiles, (file) => owners.has(file));
+    PromisePrototypeThen(SafePromiseAllReturnVoid(dependantTests, async (file) => {
+      const runningProcess = runningProcesses.get(file);
```

Comment:
We could avoid iterating over the same array twice
```suggestion
    PromisePrototypeThen(SafePromiseAllReturnVoid(testFiles, async (file) => {
      if (!owners.has(file)) return;

      const runningProcess = runningProcesses.get(file);
```